src/plugins/dashboard/server/plugin.ts

```diff
@@ -89,6 +91,51 @@ export class DashboardPlugin
 
     core.uiSettings.register(getUISettings());
 
+    const router = core.http.createRouter();
+    router.get(
+      { path: '/internal/dashboard/suggest_users', validate: false },
+      async (context, req, res) => {
+        const security = (await core.getStartServices())[1].security;
+        if (!security) {
+          return res.notFound({ body: 'Security plugin is not available' });
+        }
+
+        try {
+          const soClient = (await context.core).savedObjects.client;
+          const response = await soClient.find<
+            never,
+            { users: { buckets: Array<{ key: string }> } }
+          >({
+            type: 'dashboard',
+            perPage: 0,
+            aggs: {
+              users: {
+                terms: {
+                  field: 'created_by',
+                  size: 50,
+                },
+              },
+            },
+          });
+
+          const userProfileIds =
+            response.aggregations?.users.buckets.map((bucket) => bucket.key) ?? [];
+
+          const profiles = await security.userProfiles.bulkGet({
+            uids: new Set(userProfileIds),
+            dataPath: 'avatar',
+          });
+
+          return res.ok({
+            body: profiles,
```

Comment:
Here I simply returned `UserProfile[]`, but do we need to be concerned about backward compatibility and map this to a data transfer object?